### PR TITLE
[QuadFP][runtime] Add f90io_sc_cq_ldw for writing quad complex numbers

### PR DESCRIPTION
**Pull request by bryanpkc**

### CODE REVIEW

File: test/f90_correct/src/qc02.f08 [hunk 1]
```diff
@@ -0,0 +1,119 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! test for exponentiation operation of real complex : complex(4) ^ complex(4)
+
+! not supported!
```
Comment (pawosm-arm) [inline test/f90_correct/src/qc02.f08:7]: what are the consequences of not being supported?

File: test/f90_correct/src/precision00.f08 [hunk 2]
```diff
@@ -0,0 +1,38 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+
```
Comment (pawosm-arm) [inline test/f90_correct/src/precision00.f08:5]: single empty line (as in all other files) would do.

File: test/f90_correct/src/qp114.f08 [hunk 3]
```diff
@@ -0,0 +1,13 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! test for const quad convert to double
+program main
```
Comment (pawosm-arm) [inline test/f90_correct/src/qp114.f08:6]: please put single empty line between the comment area and the start of this program

File: test/f90_correct/src/qp125.f08 [hunk 4]
```diff
@@ -0,0 +1,73 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+
+!this test case is test for io nml
+program test
```
Comment (pawosm-arm) [inline test/f90_correct/src/qp125.f08:7]: please put single empty line between the comment area and the start of this program

File: tools/flang1/flang1exe/semutil.c [hunk 5]
```diff
@@ -713,6 +730,9 @@ cngtyp2(SST *old, DTYPE newtyp, bool allowPolyExpr)
       goto done;
 
     case TY_CMPLX:
+#ifdef TARGET_SUPPORTS_QUADFP
+    case TY_QCMPLX:
```
Comment (pawosm-arm) [inline tools/flang1/flang1exe/semutil.c:734]: shouldn't it have FLANG_FALLTHROUGH; before this case?

**Review by pawosm-arm — APPROVED** [review]
Some minor nits, but overall it's ok.

File: test/f90_correct/src/qc03.f08 [hunk 6]
```diff
@@ -0,0 +1,119 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! test for exponentiation operation of quad complex : complex(16) ^ complex(16)
+
+! not supported!
```
[the remark this answers is not in the document]
Comment (bryanpkc) [reply test/f90_correct/src/qc03.f08:7]: This was an internal "todo" comment that we forgot to remove before pushing the branch. Removed.

File: test/f90_correct/src/qc02.f08 (hunk 1 above)
Comment (bryanpkc) [reply test/f90_correct/src/qc02.f08:7]: Removed.

File: test/f90_correct/src/precision00.f08 (hunk 2 above)
Comment (bryanpkc) [reply test/f90_correct/src/precision00.f08]: Done.

File: test/f90_correct/src/qp114.f08 (hunk 3 above)
Comment (bryanpkc) [reply test/f90_correct/src/qp114.f08]: OK, I'll add a commit to format all the comments.

**Comment by bryanpkc** [comment]
@shivaramaarao Gentle ping.

**Review by shivaramaarao — APPROVED** [review]

**Pull request merged**